src/sentry/snuba/metrics/mqb_query_transformer.py

```diff
@@ -235,6 +240,139 @@ def _transform_orderby(query_orderby):
     return mq_orderby if len(mq_orderby) > 0 else None
 
 
+def _derive_mri_to_apply(project_ids, select, orderby):
+    mri_dictionary = {
+        "generic_metrics_distributions": TransactionMRI.DURATION.value,
+        "generic_metrics_sets": TransactionMRI.USER.value,
+    }
+    mri_to_apply = TransactionMRI.DURATION.value
+
+    # We first check if there is an order by field that has the team_key_transaction, otherwise
+    # we just use the default mri of duration.
+    has_order_by_tkt = False
+    if orderby is not None:
+        for orderby_field in orderby:
+            if orderby_field.field.op == TEAM_KEY_TRANSACTION_OP:
+                has_order_by_tkt = True
+                break
+
+    if has_order_by_tkt:
+        # TODO: add here optimization that gets an entity from the select (either set of distribution) and sets it
+        #  to all tkt in the query.
+        entities = set()
+        for orderby_field in orderby:
+            if orderby_field.field.op != TEAM_KEY_TRANSACTION_OP:
+                expr = metric_object_factory(orderby_field.field.op, orderby_field.field.metric_mri)
+                entities.add(expr.get_entity(project_ids, use_case_id=UseCaseKey.PERFORMANCE))
+
+        if len(entities) > 1:
+            raise InvalidParams("The orderby cannot have fields with multiple entities.")
+
+        mri_to_apply = mri_dictionary[entities.pop()]
+
+    return mri_to_apply
+
+
+def _transform_team_key_transaction_in_select(mri_to_apply, select):
+    if select is None:
+        return select
+
+    def _select(select_field):
+        if select_field.op == TEAM_KEY_TRANSACTION_OP:
+            return MetricField(
+                op=select_field.op,
+                metric_mri=mri_to_apply,
+                params=select_field.params,
+                alias=select_field.alias,
+            )
+
+        return select_field
+
+    return list(map(_select, select))
+
+
+def _transform_team_key_transaction_in_where(mri_to_apply, where):
+    if where is None:
+        return where
+
+    def _where(where_field):
```

Comment:
```suggestion
    def _where_func(where_field):
```
or another func-style idea came to my mind:
```
_where_predicate
```
but it's up to you.